### ISSUE TITLE
HKCU\Uninstall registry key created

### ISSUE THREAD

**Issue by martingbrown** [Squirrel/Squirrel.Windows#1445]
I'm not sure if this is an issue or deliberate? However the CreateUninstallerRegistryEntry method appears to be creating an 'Uninstall' sub key at the root of the HKCU hive. I suspect it was supposed to be creating 'HKCU\Software\Microsoft\Windows\CurrentVersion\Uninstall' instead.

**Comment by shiftkey** [comment]
@martingbrown are you able to provide details about what version of Squirrel you are using?

**Comment by martingbrown** [comment]
I'm on holiday at the moment so can't look out up, but the project I found this on is here https://github.com/OpenLiveWriter. I'll get a better answer later in the week.

**Comment by miloszewski2006** [comment]
@martingbrown i'm waiting for **answer**

**Comment by martingbrown** [comment]
Sorry for the slow reply, I was in hospital and forgot about this when I came out again.

The Nuget Package Version that OpenLiveWriter uses is V1.4.4. Checking the code here on GitHub I think the issue is still there.

OpenLiveWriter registers with Squirrel in its [ApplicationMain.RegisterSquirrelEventHandlers](https://github.com/OpenLiveWriter/OpenLiveWriter/blob/master/src/managed/OpenLiveWriter/ApplicationMain.cs#L206) method. In the [InitialInstall](https://github.com/OpenLiveWriter/OpenLiveWriter/blob/master/src/managed/OpenLiveWriter/ApplicationMain.cs#L246) method calls Squirrel's UpdateManager.CreateUninstallerRegistryEntry(). I'm not sure if this is the correct thing to do, I didn't actually write the code?

UpdateManager seems to simply call InstallHelpers.CreateUninstallerRegistryEntry(string,string). Starting at [line 45](https://github.com/Squirrel/Squirrel.Windows/blob/develop/src/Squirrel/UpdateManager.InstallHelpers.cs#L45) of this method there is a section of code titled "Sometimes the Uninstall key doesn't exist". 

I'm not sure why this is here. It would appear that it is trying to ensure that the  HKCU\Software\Microsoft\Windows\CurrentVersion\Uninstall registry entry exists before writing to it. However when it calls CreateSubKey it is actually creating HKCU\Unintall instead. At least that is what I think is going on.

**Comment by anaisbetts** [comment]
Yep, confirmed. Dumb typo. Doesn't seem to hurt anything, but doesn't need to be there for sure.
